dace/frontend/python/newast.py

```diff
@@ -1686,7 +1690,7 @@ def _parse_map_inputs(self, name: str, params: List[Tuple[str, str]],
                         args = ','.join([str(a) for a in expr.args])
                         if arr in self.variables:
                             arr = self.variables[arr]
-                        if arr not in self.sdfg.arrays:
+                        if not isinstance(arr, str) or arr not in self.sdfg.arrays:
```

Comment:
the reason is that arr is always a string _before_ this if, because its source is `expr.func.__name__`. After that it can become any value due to self.variables